### PR TITLE
[Thread Resource Leak] Fix thread resource leak after checkpoint catalog destroyed

### PR DESCRIPTION
**Pull request by caiconghui**
This PR is mainly to fix thread resource leak, and then add some notice to use newDaemonScheduledThreadPool api in ThreadPoolManager .

### CODE REVIEW

**Comment by caiconghui** [comment]
for #4046

File: fe/src/main/java/org/apache/doris/common/publish/FixedTimePublisher.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 public class FixedTimePublisher {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/common/publish/FixedTimePublisher.java:28]: Looks like this class is not used now.
Maybe we can mark it as @Deprecated now, or even remove it?

Comment (caiconghui) [reply fe/src/main/java/org/apache/doris/common/publish/FixedTimePublisher.java]: ok, I will remove it.

File: fe/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -458,10 +459,10 @@ public DynamicPartitionScheduler getDynamicPartitionScheduler() {
     }
 
     private static class SingletonHolder {
-        private static final Catalog INSTANCE = new Catalog();
+        private static final Catalog INSTANCE = new Catalog(false);
     }
 
-    private Catalog() {
+    private Catalog(boolean isCheckpointCatalog) {
```
Comment (WingsGo) [inline fe/src/main/java/org/apache/doris/catalog/Catalog.java:465]: maybe we should add some comment to this arg, such as: isCheckpointCatalog=true means we need to do some thread metrics collect.

Comment (caiconghui) [reply fe/src/main/java/org/apache/doris/catalog/Catalog.java:465]: ok

**Comment by morningman** [comment]
Most of UT are failed. I do not know why... Have you run it in your local env?

**Comment by caiconghui** [comment]
> Most of UT are failed. I do not know why... Have you run it in your local env?

my bad, I will fix it.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**